src/lib.rs

```diff
@@ -0,0 +1,886 @@
+	{
+		Some(connection_closed_future) => {
+			let mut connection_closed_future = Box::pin(connection_closed_future);
+			loop {
+				match futures::poll!(&mut connection_closed_future) {
+					std::task::Poll::Ready(_) => {
+						log_info!(logger, "peer connection closed: {}@{}", pubkey, peer_addr);
+						return Err(Error::ConnectionClosed);
+					}
+					std::task::Poll::Pending => {}
+				}
+				// Avoid blocking the tokio context by sleeping a bit
+				match peer_manager.get_peer_node_ids().iter().find(|id| **id == pubkey) {
+					Some(_) => return Ok(()),
+					None => tokio::time::sleep(Duration::from_millis(10)).await,
+				}
+			}
+		}
+		None => {
+			log_error!(logger, "failed to connect to peer: {}@{}", pubkey, peer_addr);
+			Err(Error::ConnectionClosed)
+		}
+	}
+}
+
+//
+// Structs wrapping the particular information which should easily be
+// understandable, parseable, and transformable, i.e., we'll try to avoid
+// exposing too many technical detail here.
+/// Represents a payment.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct PaymentInfo {
+	/// The pre-image used by the payment.
+	pub preimage: Option<PaymentPreimage>,
+	/// The secret used by the payment.
+	pub secret: Option<PaymentSecret>,
+	/// The status of the payment.
+	pub status: PaymentStatus,
+	/// The amount transferred.
+	pub amount_msat: Option<u64>,
+}
+
+/// Represents the current status of a payment.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum PaymentStatus {
+	/// The payment is still pending.
+	Pending,
+	/// The payment suceeded.
+	Succeeded,
+	/// The payment failed.
+	Failed,
+}
+
+type ChainMonitor = chainmonitor::ChainMonitor<
+	InMemorySigner,
+	Arc<dyn Filter + Send + Sync>,
+	Arc<LdkLiteChainAccess<MemoryDatabase>>,
+	Arc<LdkLiteChainAccess<MemoryDatabase>>,
+	Arc<FilesystemLogger>,
+	Arc<FilesystemPersister>,
+>;
+
+type PeerManager = SimpleArcPeerManager<
+	SocketDescriptor,
+	ChainMonitor,
+	LdkLiteChainAccess<MemoryDatabase>,
+	LdkLiteChainAccess<MemoryDatabase>,
+	dyn Access + Send + Sync,
+	FilesystemLogger,
+>;
+
+pub(crate) type ChannelManager = SimpleArcChannelManager<
+	ChainMonitor,
+	LdkLiteChainAccess<MemoryDatabase>,
+	LdkLiteChainAccess<MemoryDatabase>,
+	FilesystemLogger,
+>;
+
+type InvoicePayer<F> = payment::InvoicePayer<
+	Arc<ChannelManager>,
+	Router,
+	Arc<Mutex<Scorer>>,
+	Arc<FilesystemLogger>,
+	F,
+>;
+
+type Router = DefaultRouter<Arc<NetworkGraph>, Arc<FilesystemLogger>>;
+type Scorer = ProbabilisticScorer<Arc<NetworkGraph>, Arc<FilesystemLogger>>;
+
+type GossipSync =
+	P2PGossipSync<Arc<NetworkGraph>, Arc<dyn Access + Send + Sync>, Arc<FilesystemLogger>>;
+
+pub(crate) type NetworkGraph = gossip::NetworkGraph<Arc<FilesystemLogger>>;
+
+pub(crate) type PaymentInfoStorage = Mutex<HashMap<PaymentHash, PaymentInfo>>;
+
+pub(crate) type EventSender = mpsc::SyncSender<LdkLiteEvent>;
+pub(crate) type EventReceiver = mpsc::Receiver<LdkLiteEvent>;
+
+#[cfg(test)]
+mod tests {}
```

Comment:
lets have tests, in separate "tests" directory, instead of src files

Reply:
Good idea, added it to https://github.com/lightningdevkit/ldk-lite/issues/5.